Godeltech.Testers.Models/ThreadInfo.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Runtime.Serialization;
+using System.Xml.Serialization;
+using Godeltech.Data.Structures;
+
+namespace Godeltech.Testers.Models
+{
+    [Serializable]
+    [DataContract (Name = "thread")]
```

Comment:
When you decided to use DataContract for serialization, you should create additional DTOs and mapper for them. And apply data contracts to DTOs. 
It will protect your initial objects from specializations for serializers and attributes.